x-pack/legacy/plugins/siem/server/lib/detection_engine/routes/schemas/create_rules_schema.ts

```diff
@@ -48,8 +55,8 @@ export const createRulesSchema = Joi.object({
   rule_id,
   index,
   interval: interval.default('5m'),
-  query: query.allow('').default(''),
-  language: language.default('kuery'),
+  query, // TODO conditional type/default?
+  language,
```

Comment:
I would utilize conditionals for `language` and `query` if you want them to disappear when you have a type of `machine_learning`. As this is written right now, it won't be backwards compatible and begin breaking existing rules and code that do not have `language` set or `query` not set.

For example these ad-hoc tests:

```ts
cat ./rules/queries/query_with_rule_id.json 
{
  "name": "Query with a rule id",
  "description": "Query with a rule_id that acts like an external id",
  "rule_id": "query-rule-id",
  "risk_score": 1,
  "severity": "high",
  "type": "query",
  "query": "user.name: root or user.name: admin"
}
```

You can use the `when` for when it is machine_learning. You can optionally just leave the default event though you are not using it within the code even though it will default that value to be `kquery`.

You can also set the language to be `machine_learning` as well when your type to `machine_learning` or to be even something like `algorithm` when nothing is specified and your type is `machine_learning`. We don't expect multiple language types for machine learning but if you just want to be consistent you could do that optionally as well.